docs/src/modules/utils/useLazyCSS.ts

```diff
@@ -2,16 +2,71 @@ import * as React from 'react';
 import { loadCSS } from 'fg-loadcss';
 
 /**
- * Convenience wrapper around fgLoadCSS for hooks usage
- * @param {string} href
- * @param {string} before - CSS selector
+ * Enhanced lazy CSS loader that wraps CSS in a layer using fetch to avoid CORS issues
+ * @param {string} href - URL of the CSS file to load
+ * @param {string} before - CSS selector to insert before
+ * @param {object} options - Additional options
+ * @param {string} options.layer - Optional CSS layer name to wrap the CSS in
  * @returns {() => void} cleanup function
  */
-export default function useLazyCSS(href: string, before: string) {
+export default function useLazyCSS(href: string, before: string, options: { layer?: string } = {}) {
   React.useEffect(() => {
-    const link = loadCSS(href, document.querySelector(before) as HTMLElement);
+    // If no layer is specified, use the original behavior
+    if (!options.layer) {
+      const link = loadCSS(href, document.querySelector(before) as HTMLElement);
+      return () => {
+        link.parentElement?.removeChild(link);
+      };
+    }
+
+    // With layer option, we need to fetch the CSS content and wrap it
+    let styleElement: HTMLStyleElement | null = null;
+    const abortController = new AbortController();
+
+    // Fetch the CSS content directly to avoid CORS issues with cssRules
```

Comment:
I don't understand how we could get CORS issues. Everything comes from the same origin 🤔

But more security do not hurt